### ISSUE TITLE
set_tld_regex always causes modSettings cache to be invalid

### ISSUE THREAD

**Issue by jdarwood007** [SimpleMachines/SMF#3702]
This took me hours to figure out why modSettings kept getting a cache miss despite having a 90 second ttl.  Finally traced it back.

In Subs.php at around lines 963 we have
````
	// Ensure $modSettings['tld_regex'] contains a valid regex for the autolinker
	if (!empty($modSettings['autoLinkUrls']))
		set_tld_regex();
````

In the set_tld_regex we have this to start
````
function set_tld_regex($update = false)
{
	global $sourcedir, $smcFunc;
	static $done = false;
````

And then later on
````
	// If we aren't updating and the regex is valid, we're done
	elseif (!empty($modSettings['tld_regex']) && @preg_match('~' . $modSettings['tld_regex'] . '~', null) !== false)
```

Because modSettings isn't defined, we fail the test and the system falls back to going to the default tlds and then issues a updateSettings, which causes a invalidation of the cache.

````
	// Remember the new regex in $modSettings
	updateSettings(array('tld_regex' => $tld_regex));
````

**Comment by jdarwood007** [comment]
@Sesquipedalian

**Comment by Sesquipedalian** [comment]
Derp. Fixed in #3704
